qemu/tests/vsock_seqpacket_test.py

```diff
@@ -0,0 +1,97 @@
+import logging
+import random
+import time
+import aexpect
+
+from avocado.utils import process
+
+from virttest import error_context
+from virttest import utils_misc
+from virttest import utils_net
+
+LOG_JOB = logging.getLogger('avocado.test')
+
+
+def download_compile_upstream_testsuite(test, params, session):
+    """
+    Download and compile upstream test suite on guest and host.
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param session: Guest session object
+    """
+    clone_cmd = params["clone_cmd"]
+    host_status = process.system(clone_cmd, timeout=720, shell=True)
+    guest_status = session.cmd_status(clone_cmd, timeout=1800)
+    if (host_status or guest_status) != 0:
+        test.error("Test suite download failed")
+
+    compile_cmd = params["compile_cmd"]
+    host_status = process.system(compile_cmd, shell=True)
+    guest_status = session.cmd_status(compile_cmd)
+    if (host_status or guest_status) != 0:
+        test.error("Test suite compile failed")
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Vsock seqpacket test
+
+    1. Boot guest with vhost-vsock-pci device
+    2. Disable firewall in guest
+    3. Download and compile upstream test suite in guest and host
+    4. Run test suite
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment
+    """
+    vm = env.get_vm(params["main_vm"])
+    vm.verify_alive()
+    session = vm.wait_for_login()
+    host_ip = utils_net.get_host_ip_address(params)
+    guest_ip = vm.get_address()
+
+    disable_firewall = params.get("disable_firewall")
+    session.cmd(disable_firewall, ignore_all_errors=True)
+
+    vsock_dev = params["vsocks"].split()[0]
+    guest_cid = vm.devices.get(vsock_dev).get_param("guest-cid")
+    host_cid = params["host_cid"]
+    port = random.randrange(1, 6000)
+    test_bin = params["testsuite_dir"] + "vsock_test"
+
+    try:
+        download_compile_upstream_testsuite(test, params, session)
+
+        # Scenario I: host = client, guest = server
+        LOG_JOB.info("Host as client, guest as server...")
```

Comment:
Should use `test.log` in the `run` function.

Reply:
Updated.